projects/tier-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/tier/TierService.kt

```diff
@@ -0,0 +1,116 @@
+import uk.gov.justice.digital.hmpps.integrations.delius.contact.type.ContactTypeCode
+import uk.gov.justice.digital.hmpps.integrations.delius.contact.type.ContactTypeRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.contact.type.getByCode
+import uk.gov.justice.digital.hmpps.integrations.delius.management.ManagementTier
+import uk.gov.justice.digital.hmpps.integrations.delius.management.ManagementTierId
+import uk.gov.justice.digital.hmpps.integrations.delius.management.ManagementTierRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.person.Person
+import uk.gov.justice.digital.hmpps.integrations.delius.person.PersonRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.person.getByCrnAndSoftDeletedIsFalse
+import uk.gov.justice.digital.hmpps.integrations.delius.referencedata.ReferenceData
+import uk.gov.justice.digital.hmpps.integrations.delius.referencedata.ReferenceDataRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.referencedata.getByCodeAndSetName
+import uk.gov.justice.digital.hmpps.integrations.delius.staff.StaffRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.staff.getByCode
+import uk.gov.justice.digital.hmpps.integrations.delius.team.TeamRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.team.getByCode
+import uk.gov.justice.digital.hmpps.telemetry.TelemetryService
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+
+@Service
+class TierService(
+    private val tierClient: TierClient,
+    private val personRepository: PersonRepository,
+    private val referenceDataRepository: ReferenceDataRepository,
+    private val managementTierRepository: ManagementTierRepository,
+    private val contactRepository: ContactRepository,
+    private val staffRepository: StaffRepository,
+    private val teamRepository: TeamRepository,
+    private val contactTypeRepository: ContactTypeRepository,
+    private val telemetryService: TelemetryService,
+) {
+    fun handleTierCalculation(crn: String, calculationId: String) {
+        val tierCalculation = tierClient.getTierCalculation(crn, calculationId)
+        updateTier(crn, tierCalculation)
+        telemetryService.trackEvent(
+            "TierUpdateSuccess",
+            mapOf(
+                "crn" to crn,
+                "tier" to tierCalculation.tierScore,
+                "calculationDate" to tierCalculation.calculationDate.toString()
+            )
+        )
+    }
+
+    @Transactional
+    fun updateTier(crn: String, tierCalculation: TierCalculation) {
+        val tier = referenceDataRepository.getByCodeAndSetName("U${tierCalculation.tierScore}", "TIER")
+        val person = personRepository.getByCrnAndSoftDeletedIsFalse(crn)
+        val changeReason = referenceDataRepository.getByCodeAndSetName("ATS", "TIER CHANGE REASON")
+
+        createTier(person, tier, tierCalculation.calculationDate, changeReason)
```

Comment:
Good point, cheers - we'll have a look tomorrow

Reply:
Had a look through the code and it turns out Community API doesn't do any checks against the current tier value, so it always saves a new entry for each message (unless I'm missing something!)

However, this got me thinking that the `person.currentTier` update should only happen for the most recent tier update - in case we receive messages out of order.  So there may be a check we need to do there - will look at that today.

Reply:
Serves me right for not pulling the latest!  Not sure when that was added but it appears my local was wildly out of date 🤦‍♂️ 
![image](https://user-images.githubusercontent.com/39557241/195553487-b0501a9b-61de-4153-a531-8c2acee7f4da.png)